packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -1546,3 +1569,43 @@ class _ReorderableItemGlobalKey extends GlobalObjectKey {
   @override
   int get hashCode => Object.hash(subKey, index, state);
 }
+
+/// [ReorderableDragBoundary] is a widget that limits the drag boundary of a [ReorderableList].
+///
+/// When this widget wraps a [ReorderableList], the drag boundary of the [ReorderableList] is confined within this widget.
+/// If the [dragBoundingRect] property is provided, the drag boundary is limited to the rectangle returned by [dragBoundingRect].
+/// The boundary is specified in global coordinates.
+class ReorderableDragBoundary extends InheritedWidget {
+  /// Create a [ReorderableDragBoundary] to set the drag boundary for the [ReorderableList].
+  ///
+  /// The [dragBoundingRect] needs to return a rectangle in global coordinates.
+  const ReorderableDragBoundary({
+      required super.child,
+      this.dragBoundingRect,
```

Comment:
This feels a bit difficult for users to define. Say the user wants to stick within the bounds of the Viewport, since a Viewport will expand to fill all of the available space, how does the user know what that rect is?